### PR TITLE
feat: document types improvements

### PR DESCRIPTION
**Pull request by ablause**
Someone would help me to check if everything works?

Features:
- Improvements in the types of filters and inserts.
- Added query, update and pagination operators. 

Fixes: #237 #183

### CODE REVIEW

**Comment by erfanium** [comment]
Great job!
I'll check when it's green

**Comment by ablause** [comment]
I have some concerns about my projects with this PR, I'll fix it right away..

**Comment by manyuanrong** [comment]
@ablause Nice work, do you still need to make other changes?

**Comment by ablause** [comment]
@manyuanrong Yes, I'm finishing the last changes today

File: src/collection/collection.ts [hunk 1]
```diff
@@ -130,23 +142,26 @@ export class Collection<T> {
...
 
   async insertMany(
-    docs: Document[],
+    docs: InsertDocument<T>[],
     options?: InsertOptions,
   ): Promise<{ insertedIds: Document[]; insertedCount: number }> {
     const insertedIds = docs.map((doc) => {
       if (!doc._id) {
-        doc._id = new Bson.ObjectID();
+        (doc as any)._id = new Bson.ObjectID();
```
Comment (erfanium) [inline src/collection/collection.ts:164]:
```suggestion
        (doc as any)._id = new Bson.ObjectId();
```

To fix conflict

**Review by erfanium — CHANGES_REQUESTED** [review]

**Comment by ablause** [comment]
I tried to fill in as much as possible, I hope I didn't make too many mistakes.  The operators value for the aggregations are missing, if one day I have the courage to do it, I'll do a pull-request especially for this one

**Pull request merged**